WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -69,13 +69,14 @@ def select_nearby_delegate
 
   def edit_avatar_thumbnail
     @user = user_to_edit
-    return if redirect_unless_user(:can_change_users_avatar?, @user)
+    redirect_to_root_unless_user(:can_change_users_avatar?, @user)
```

Comment:
I'm not in love with the `return` getting dropped here, because I worry about code getting added below this line. That said, if you prefer it this way, I'm okay with that. No need to rehash the discussion, I just had to put that comment here for anyone watching.